File: frontend/src/services/vipPackageService.js
import api from '../api/axiosConfig';

export const vipPackageService = {
  // Public APIs
  getActivePackages: () => api.get('/api/vip-packages'),

  // Admin APIs
  getAllAdminPackages: () => api.get('/api/admin/vip-packages'),

  getPackageById: (id) => api.get(`/api/admin/vip-packages/${id}`),

  createPackage: (data) => api.post('/api/admin/vip-packages', data),

  updatePackage: (id, data) => api.put(`/api/admin/vip-packages/${id}`, data),

  togglePackageStatus: (id) => api.patch(`/api/admin/vip-packages/${id}/toggle`),

  deletePackage: (id) => api.delete(`/api/admin/vip-packages/${id}`),
};

File: backend/src/main/java/com/storyapp/storyapp/service/VipPackageService.java
package com.storyapp.storyapp.service;

import com.storyapp.storyapp.dto.request.VipPackageRequest;
import com.storyapp.storyapp.dto.response.VipPackageResponse;

import java.util.List;

public interface VipPackageService {

    List<VipPackageResponse> getActivePackages();

    List<VipPackageResponse> getAllPackages();

    VipPackageResponse getPackageById(Long id);

    VipPackageResponse createPackage(VipPackageRequest request);

    VipPackageResponse updatePackage(Long id, VipPackageRequest request);

    VipPackageResponse togglePackageStatus(Long id);

    void deletePackage(Long id);
}
